core/src/main/scala/org/http4s/StaticFile.scala

```diff
@@ -87,15 +87,24 @@ object StaticFile {
             else `Transfer-Encoding`(TransferCoding.chunked)
           val headers = Headers(lenHeader :: lastModHeader ::: contentType)
 
-          Some(
-            Response(
-              headers = headers,
-              body = readInputStream[F](F.delay(url.openStream), DefaultBufferSize, blocker)
-            ))
-        } else {
-          urlConn.getInputStream.close()
-          Some(Response(NotModified))
-        }
+          F.delay(url.openStream)
+            .redeem(
+              recover = {
+                case _: FileNotFoundException => None
+                case other => throw other
+              },
+              f = { inputStream =>
```

Comment:
Was struggling a bit to understand what is `f` here? Then realized that it is just the `redeem` function's parameter name :)